docs/messaging/rabbitmq-client-integration.md

```diff
@@ -1,104 +1,214 @@
+When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `docker.io/library/rabbitmq` image, it creates a new RabbitMQ server instance on your local machine. A reference to your RabbitMQ server (the `rabbitmq` variable) is added to the `ExampleProject`. The RabbitMQ server resource defaults include a `username` of `"guest"` and a `password` that's randomly generated using the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter*>
+
+The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `"messaging"`.
+
+> [!TIP]
+> If you'd rather connect to an existing RabbitMQ server, call <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString*> instead. For more information, see [Reference existing resources](../fundamentals/app-host-overview.md#reference-existing-resources).
+
+### Add RabbitMQ server resource with parameters
+
+When you want to explicitly provide the username and password used by the container image, you can provide those as parameters. Consider the following alternative example:
 
 ```csharp
-public class ExampleService(IConnection connection)
-{
-    // Use connection...
-}
+var builder = DistributedApplication.CreateBuilder(args);
+
+var username = builder.AddParameter("username", secret: true);
+var password = builder.AddParameter("password", secret: true);
+
+var rabbitmq = builder.AddRabbitMQ("messaging", username, password);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(rabbitmq);
+
+// After adding all resources, run the app...
+```
+
+For more information on providing parameters, see [External parameters](../fundamentals/external-parameters.md).
+
+### Add RabbitMQ server resource with management plugin
+
+To add the [RabbitMQ management plugin](https://www.rabbitmq.com/docs/management) to the RabbitMQ server resource, call the <xref:Aspire.Hosting.RabbitMQBuilderExtensions.WithManagementPlugin*> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var rabbitmq = builder.AddRabbitMQ("messaging")
+                      .WithManagementPlugin();
+
+builder.AddProject<Projects.ExampleProject>()
+        .WithReference(rabbitmq);
+
+// After adding all resources, run the app...
+```
+
+The RabbitMQ management plugin provides an HTTP-based API for management and monitoring of your RabbitMQ server. .NET Aspire will add another container image `docker.io/library/rabbitmq-management` to the app host that runs the management plugin.
+
+### Add RabbitMQ server resource with data volume
+
+To add a data volume to the RabbitMQ server resource, call the <xref:Aspire.Hosting.RabbitMQBuilderExtensions.WithDataVolume*> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var rabbitmq = builder.AddRabbitMQ("messaging")
+                      .WithDataVolume("/var/lib/rabbitmq");
```

Comment:
```suggestion
                      .WithDataVolume();
```

This string is the volume name, and it is optional.

Reply:
Good to know, I'll remove it then and maybe add a note that when not explicitly passing in a path for the volume, it defaults to that.